source/val/validate_extensions.cpp

```diff
@@ -2962,12 +2962,29 @@ spv_result_t ValidateExtInst(ValidationState_t& _, const Instruction* inst) {
                  << "expected operand Format to be a pointer";
         }
 
-        if (format_storage_class != spv::StorageClass::UniformConstant) {
+        if (format_storage_class != spv::StorageClass::UniformConstant &&
+            // Extension SPV_EXT_relaxed_printf_string_address_space allows
+            // format strings in Global, Local, Private and Generic address
+            // spaces
+
+            // Global
```

Comment:
As these are only valid if the extension is enabled, it would be better to only allow them if the extension is actually enabled with `OpExtension`.

Reply:
> As these are only valid if the extension is enabled, it would be better to only allow them if the extension is actually enabled with `OpExtension`.

I tried to check with:
`_.HasExtension(Extension::kSPV_EXT_relaxed_printf_string_address_space)`
but Extension::kSPV_EXT_relaxed_printf_string_address_space is not available.
'SPV_EXT_relaxed_printf_string_address_space' is in SPIRV-LLVM-Translator/include/LLVMSPIRVExtensions.inc, but it is not in SPIRV-Headers.  What is the correct way to make it available in SPIRV-Tools?

Reply:
Not entirely sure if this is the solution, but worth a try: does it work if you add the extension to `EXTENSIONS_FROM_SPIRV_REGISTRY_AND_NOT_FROM_GRAMMARS` in `utils/generate_grammar_tables.py`?  I guess it doesn't come automatically from the grammar files because there is no instruction that requires the extension.

Reply:
Yes, adding the extension to   `EXTENSIONS_FROM_SPIRV_REGISTRY_AND_NOT_FROM_GRAMMARS` does work.  

Should we go with this solution?

Reply:
Thanks, LGTM but it would be great to get an approval from someone more familiar with the code.